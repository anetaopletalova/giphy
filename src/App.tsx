import React from 'react'
import {BrowserRouter, Routes, Route} from 'react-router-dom'
import GiphyApp from './components/GiphyApp/GiphyApp'

const App: React.FC = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<GiphyApp />} />
      </Routes>
    </BrowserRouter>
  )
}

export default App
