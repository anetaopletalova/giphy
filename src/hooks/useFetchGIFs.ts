import {useQuery} from 'react-query'
import {getGIFS} from '../api'

export const useFetchGIFs = (
  q: string | null,
  limit: string,
  offset: string,
) => {
  const {data, isLoading, isError} = useQuery(
    'GIFs',
    () => getGIFS('s', limit, offset),
    {enabled: !!q},
  )

  return {data, isLoading, isError}
}
