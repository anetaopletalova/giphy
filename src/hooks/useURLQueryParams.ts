import {useMemo} from 'react'
import {useLocation, useSearchParams} from 'react-router-dom'

export const useURLQueryParams = () => {
  const [searchParams, setSearchParams] = useSearchParams()
  const {search} = useLocation()
  const query = useMemo(() => new URLSearchParams(search), [search])

  const q = query.get('q') || ''
  const limit = query.get('limit') || '20'
  const offset = query.get('offset') || '0'

  const setParams = (param: string, value: string) => {
    searchParams.set(param, value)
    setSearchParams(searchParams.toString())
  }

  return {q, limit, offset, setParams}
}
