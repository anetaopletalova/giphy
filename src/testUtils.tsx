import React from 'react'
import {QueryClient, QueryClientProvider} from 'react-query'
import {render} from '@testing-library/react'

const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })

export const renderWithClient = (children: React.ReactElement) => {
  const queryClient = createTestQueryClient()
  const {rerender, ...result} = render(
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>,
  )
  return {
    ...result,
    rerender: (rerenderUi: React.ReactElement) =>
      rerender(
        <QueryClientProvider client={queryClient}>
          {rerenderUi}
        </QueryClientProvider>,
      ),
  }
}
