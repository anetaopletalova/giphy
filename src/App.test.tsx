import {fireEvent} from '@testing-library/react'
import React from 'react'
import App from './App'
import {useFetchGIFs} from './hooks/useFetchGIFs'
import {renderWithClient} from './testUtils'

jest.mock('./hooks/useFetchGIFs', () => ({
  useFetchGIFs: jest.fn(),
}))

describe('Giphy App', () => {
  beforeEach(() => {
    ;(useFetchGIFs as any).mockImplementation(() => {})
  })
  it('shows gif container on search', async () => {
    const rendered = renderWithClient(<App />)

    const searchInput = rendered.getByTestId('search').querySelector('input')
    expect(searchInput).toBeInTheDocument()

    fireEvent.change(searchInput as HTMLInputElement, {
      target: {value: 'dogs'},
    })
    expect((searchInput as HTMLInputElement).value).toBe('dogs')
    await new Promise((r) => setTimeout(r, 3500))
    expect(rendered.container).toMatchSnapshot()

    expect(rendered.getByTestId('gif-container')).toBeInTheDocument()
  })
})
