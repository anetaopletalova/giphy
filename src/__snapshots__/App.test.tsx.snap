// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Giphy App shows gif container on search 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="container"
    >
      <span
        class="title"
      >
        Let's search for some GIFs!
      </span>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        data-testid="search"
        style="width: 350px;"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for=":r0:"
          id=":r0:-label"
        >
          GIF name
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            maxlength="50"
            type="text"
            value="dogs"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                GIF name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        data-testid="gif-container"
        style="display: flex; flex-direction: column; justify-content: end;"
      >
        <div
          class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
              id=":r2:"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":r2: :r1:"
                class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                id=":r1:"
                role="button"
                tabindex="0"
              >
                20
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="20"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
            >
              1â€“20 of 15106
            </p>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to previous page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </button>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                title="Go to next page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-12n0uxm-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Dog GIF by ViralHog"
              loading="lazy"
              src="https://media1.giphy.com/media/lNJWuuYxPUTbPIF9id/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Well Done Applause GIF by MOODMAN"
              loading="lazy"
              src="https://media1.giphy.com/media/kBZBlLVlfECvOQAVno/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Dog GIF by ViralHog"
              loading="lazy"
              src="https://media0.giphy.com/media/3ocB9c8W45am8rfSin/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Wake Up Dog GIF"
              loading="lazy"
              src="https://media1.giphy.com/media/9s8Jq3Sc1ZnZS/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Dogs Puppies GIF by MOODMAN"
              loading="lazy"
              src="https://media2.giphy.com/media/Z9tvqoD1SEQcomTVaK/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Animal Planet Dog GIF by Puppy Bowl"
              loading="lazy"
              src="https://media3.giphy.com/media/l0NwCwLcizTe8bFIc/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="digging wiener dog GIF by Giatec Scientific"
              loading="lazy"
              src="https://media4.giphy.com/media/3ov9jRPMChw9ZzVlUk/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Dog GIF by ViralHog"
              loading="lazy"
              src="https://media2.giphy.com/media/tXG1bTpN7uz839gSZA/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Golden Retriever Hello GIF by Finley"
              loading="lazy"
              src="https://media3.giphy.com/media/If0uerEAtNFl8FpNw4/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Dog GIF by ViralHog"
              loading="lazy"
              src="https://media4.giphy.com/media/ly496ILvrTxnMgBBYM/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="I Love You Dogs GIF by Chippy the Dog"
              loading="lazy"
              src="https://media2.giphy.com/media/kFt36Dvf7xLZuNQbA4/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Oh Yeah Ok GIF by ScalerFab"
              loading="lazy"
              src="https://media4.giphy.com/media/qDstf5cc8JsKRo0IlZ/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Confused Dogs GIF by MOODMAN"
              loading="lazy"
              src="https://media1.giphy.com/media/xT0xeuOy2Fcl9vDGiA/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Star Wars Dogs GIF"
              loading="lazy"
              src="https://media3.giphy.com/media/fItgT774J3nWw/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Drunk Dog GIF"
              loading="lazy"
              src="https://media3.giphy.com/media/11FOb5AeCLT8oo/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Cops And Robbers Dogs GIF by reactionseditor"
              loading="lazy"
              src="https://media0.giphy.com/media/3ohzdYGKrPn8GzgAes/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Help Me Please GIF"
              loading="lazy"
              src="https://media0.giphy.com/media/Or6zGCW7Rdk45o9I2c/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Dogs Driving GIF"
              loading="lazy"
              src="https://media1.giphy.com/media/KEh5kliRTSVJm/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Bored Brown Dog GIF"
              loading="lazy"
              src="https://media1.giphy.com/media/mNLh4BQKRrNQbFbuCF/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Dog Love GIF by Pamily"
              loading="lazy"
              src="https://media3.giphy.com/media/l2SpMU9sWIvT2nrCo/200.gif?cid=87181defe95sh3uaq2zetpnd50wy25e8ttybeeb8kixlrlyv&rid=200.gif&ct=g"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
